sql/core/src/test/scala/org/apache/spark/sql/DataFrameWriterV2Suite.scala

```diff
@@ -531,6 +534,23 @@ class DataFrameWriterV2Suite extends QueryTest with SharedSparkSession with Befo
     assert(table.properties === (Map("provider" -> "foo") ++ defaultOwnership).asJava)
   }
 
+  test("SPARK-39543 writeOption should be passed to storage properties when fallback to v1") {
+    val provider = classOf[InMemoryV1Provider].getName
+
+    withSQLConf((SQLConf.USE_V1_SOURCE_LIST.key, provider)) {
```

Comment:
looking at other tests that use `InMemoryV1Provider`, it seems we don't need to set `USE_V1_SOURCE_LIST` to trigger v1 fallback?

Reply:
Yea, Other tests trigger v1 fallback without set `USE_V1_SOURCE_LIST `, AFAIK, 
- Other tests aim to test the read/write process, and the `InMemoryV1Provider` is actually a v2 format, and we trigger v1 fallback at the `newScanBuilder` & `newWriteBuilder` layer.
- This test in PR needs to be fallback to V1 when the table is created, so we need to set `USE_V1_SOURCE_LIST`(see: [isV2Provider](https://github.com/apache/spark/blob/3d68ad8003a16229bd79f86cb31f618167814a7f/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L604))